### DISCUSSION TITLE
20240530 Weekly Development Report

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#96] [Meetings]
There is no meeting this week. We can just write down relevant tasks, questions, etcetera.

- Cameron Garnham(@da2ce7)
- Jose Celano(@josecelano)
- Constantin Bosse(@cgbosse)
- Mario Melchor(@mario-nt)
- Graeme Byrne(@grmbyrn)
- Borja (@BorjaPractica)

**Comment by josecelano** [comment]
### Torrust Index GUI

Issues and PRs:

- https://github.com/torrust/torrust-index-gui/issues/562
- https://github.com/torrust/torrust-index-gui/pull/565

### Torrust Index

Issues:

I've continued with the configuration overhaul.

- https://github.com/torrust/torrust-index/issues/599
- https://github.com/torrust/torrust-index/issues/603
- https://github.com/torrust/torrust-index/issues/605

@da2ce7 I consider it done. I have postponed breaking changes in config files for the next release. See:

- https://github.com/torrust/torrust-index/issues/591
- Relates to: https://github.com/torrust/torrust-tracker/issues/878

PR review:

- https://github.com/torrust/torrust-tracker/pull/876

### Torrust Compose

Issues:

- https://github.com/torrust/torrust-compose/issues/14
- https://github.com/torrust/torrust-compose/issues/3

### Torrust Website

Issues and PRs:

- https://github.com/torrust/torrust-website/issues/114
- https://github.com/torrust/torrust-website/issues/121
- https://github.com/torrust/torrust-website/issues/138
- https://github.com/torrust/torrust-website/issues/161
- https://github.com/torrust/torrust-website/pull/187

### Operative

- Review [roadmap](https://github.com/orgs/torrust/projects/10/views/6) (v3.0.0) with @cgbosse. Hey @da2ce7 take a look at it. We are close to finishing all the issues @cgbosse and I have included in the 3.0.0 milestone. Tell us if you miss something important.
- Evaluation of interns: Ismael and Iván. They finished their internship.

### Research

I've installed https://www.ox.security/ in our main repos. Not many new security issues were detected:

- https://github.com/torrust/torrust-index/issues/610
- https://github.com/torrust/torrust-index-gui/issues/566

### Courses

Tech Lead Course by @flopezluis

### Events

Meetup: [Approval Testing](https://www.meetup.com/gofigeeks/events/300998862/)

**Comment by mario-nt** [comment]
### Index
	
- https://github.com/torrust/torrust-index/issues/448 -> Final refactor and working on fixing merge conflicts when rebasing with the current develop branch
- https://github.com/torrust/torrust-tracker/issues/670 -> Refactor logic to be implemented with traits

- https://github.com/torrust/torrust-index/pull/609 -> Update deps

**Comment by grmbyrn** [comment]
### Website
* [Add search bar, fix storybook, fix other minor errors #186](https://github.com/torrust/torrust-website/pull/186) -> Create and add `SearchBar` component to `Header` component to search for key words in title, excerpt and tags of blog posts.

* [Running the storybook in dev mode does not work #104](https://github.com/torrust/torrust-website/issues/104) -> Updated dependencies to fix storybook errors.

* [Missing favicon.ico #160](https://github.com/torrust/torrust-website/issues/160) -> Corrected file path to correctly access and display favicon.

* [Build error: 404 (favicon.ico, undefined, links) #167](https://github.com/torrust/torrust-website/issues/167) -> Corrected links in blog posts table of contents so that they correctly linked to the correct section.
